src/libchdr_chd.c

```diff
@@ -1491,6 +1544,11 @@ static inline void map_assemble(uint8_t *base, map_entry *entry)
 -------------------------------------------------*/
 static inline int map_size_v5(chd_header* header)
 {
+	// Avoid overflow due to corrupted data.
+	const uint32_t max_hunkcount = (UINT32_MAX / header->mapentrybytes);
+	if (header->hunkcount > max_hunkcount)
```

Comment:
This seems to be the sole remaining newly introduced comparison that doesn't have the variable on the left side.

Reply:
The variable from the file is on the left in this case, `max_hunkcount <= header->hunkcount` definitely does not read correctly to me.